src/main/java/seedu/address/model/order/OrderTracker.java

```diff
@@ -3,13 +3,13 @@
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Objects;
 
 /**
  * Track all the order history of a customer
  */
 public class OrderTracker {
     private ArrayList<OrderHistory> history;
+    /** Internal use only, frequency should be the same iff history is the same */
     private HashMap<Order, Integer> frequency;
 
     /**
```

Comment:
Could you perhaps use requireNonNull in constructor to enforce defensive coding?